src/devices/Nrf24l01/Nrf24l01.cs

```diff
@@ -0,0 +1,773 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Threading;
+
+namespace Iot.Device.Nrf24l01
+{
+    /// <summary>
+    /// Single chip 2.4 GHz Transceiver nRF24L01
+    /// </summary>
+    public class Nrf24l01 : IDisposable
+    {
+        private GpioController _gpio = null;
+        private SpiDevice _sensor = null;
+
+        private readonly int _ce;
+        private readonly int _irq;
+
+        private readonly byte[] _empty = new byte[0];
+
+        #region prop
+        private byte _packetSize;
+        /// <summary>
+        /// nRF24L01 Receive Packet Size
+        /// </summary>
+        public byte PacketSize
+        {
+            get { return _packetSize; }
+            set { _packetSize = value < 0 || value > 32 ? throw new ArgumentOutOfRangeException("PacketSize needs to be in the range of 0 to 32!") : value; }
+        }
+
+        /// <summary>
+        /// nRF24L01 Send Address
+        /// </summary>
+        public Span<byte> Address { get => ReadTxAddress(); set => SetTxAddress(value); }
```

Comment:
I think this should be `void ReadAddress(Span<byte> address)` or `byte[] Address` returning span only makes sense when it is a subspan of the input argument